Automattic/newspack-newsletters
# fix: always use cache to retrieve groups

### All Submissions:






### Changes proposed in this Pull Request:



In https://github.com/Automattic/newspack-newsletters/pull/1063 we introduced a non obstructive caching for fetching the groups that are displayed in the "send to" dropdown menu, and it works fine :) 

But then in https://github.com/Automattic/newspack-newsletters/pull/1156 we started fetching for groups while getting the lists. And we were not using the same caching strategy here.

Not only it makes the UI slow, it also affects the speed of the dropdown that was fixed in #1063, because that API request ends up calling `get_lists` via the `retrieve()` method.

This PR makes the `get_lists` also use the cache.

Also. it increases the limit of group categories and groups retrieved by the api request

### How to test the changes in this Pull Request:

1. This affects only Mailchimp, so set Maiclhimp as the ESP
2. Make sure the SEND TO dropdown is working fine, displaying the group categories and groups
3. Make sure Newsletters > Settings lists the Lists correctly, and also Newspack > Engagement > Newsletters
4. Make sure subscription blocks work as expected
5. Basically make sure things still work :) 


### Other information:





---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205719582509600
  - https://app.asana.com/0/0/1206084407862419
